editoast/src/main.rs

```diff
@@ -116,11 +117,21 @@ fn init_tracing(mode: EditoastMode, telemetry_config: &client::TelemetryConfig)
             Some(layer)
         }
     };
+
+    if for_test {
+        let subscriber = tracing_subscriber::registry()
+            .with(telemetry_layer)
+            .with(env_filter_layer)
+            .with(fmt_layer);
+        return Some(tracing::subscriber::set_default(subscriber));
+    }
```

Comment:
Did you try to entirely remove that? [`set_default` set a subscriber only for the current scope](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/util/trait.SubscriberInitExt.html#method.set_default). But below, we use `.init()` which will globally set up the subscriber. Do we need a subscriber per-test? Maybe it's because `.init()` will panic if call multiple time in the same process and since, all tests are launch in parallel in the same process, it doesn't work?

In any case, I believe it should be possible to keep the code identical for production code and test code. I see 2 options.

1. Either we use `set_default` everywhere and we keep the guard in the `run()` function as well as in `TestApp` so the subscriber lives for the entire application (but as mentioned [here](https://docs.rs/tracing/0.1.21/tracing/dispatcher/index.html#setting-the-default-subscriber), it will not work well with `thread::spawn`, I wonder about `tokio::spawn`?). 
2. We use `try_init` instead of `init()`. It should work the same for production code, and for tests, only the first test will initialize the subscriber.

One last thing, I believe we don't need `for_test` variable, we can use `#[cfg(test)]` to detect if we're in tests or not.

Reply:
Yes, you're right. The issue comes from calling `.init()` multiple times in the same process, which panics when used in parallel tests.
Option 2 works great and solved the issue for me. Thanks a lot for the suggestion! It’s much cleaner and avoids the panic. Appreciate the help!

Reply:
Reading a bit more the documentation, I realize that we might set up a global subscriber, then also have per-scope subscriber. In this case, [the global subscriber will only serve as a fall-back when a scoped one is not found](https://docs.rs/tracing/latest/tracing/dispatcher/fn.set_global_default.html).

I'm not entirely sure on the effect of having only one global subscriber for all the tests. Maybe we should have one scoped subscriber for each test? Therefore, maybe the subscriber itself should be returned by the function? the `run()` would just `.init()` (or `tracing::dispatcher::set_global_default(subscriber.into())`, it's equivalent). `TestApp` would use `set_default()`. What do you think? Maybe @leovalais can give some insights about why it was scoped in the first place with `DefaultGuard` (it was introduced in 35b9369ebc8d7fba1637c2c0f50f7d7f26db8536 of https://github.com/OpenRailAssociation/osrd/pull/7511)?

Reply:
Then I would change the way I proposed above, the function `init_tracing` should not be in charge of installing (globally or not) the subscriber, it should only return it. Then `.run()` will be able to set it up with `.init()`. And `TestApp` will be able to use `set_default()` and keep the `DefaultGuard`. I feel it would be a better design. What do you think?

Reply:
Changes made as per your suggestion. Please check the code. Thanks again! @woshilapin @leovalais